paasta_tools/cli/cmds/local_run.py

```diff
@@ -754,6 +769,29 @@ def run_docker_container(
         healthcheck_only or healthcheck
     ) and healthcheck_mode is not None
 
+    for container_mount_path, secret_content in secret_volumes.items():
+        temp_secret_folder = tempfile.TemporaryDirectory(
```

Comment:
When having many volumes, I suspect there is the risk of this variable being garbage collected, and consequently the directory deleted.

Reply:
huh, yeah I had thought about this, based on the [documentation](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory):
> On completion of the context or destruction of the temporary directory object, the newly created temporary directory and all its contents are removed from the filesystem.

But I didn't observe that in practice in my local testing, but I hadn't considered the "when having many volumes" part.

Reply:
@mpiano++ great call here. 

```
  secret_volumes:
  - container_path: /who/config/
    secret_name: config_yaml_file
    default_mode: '0444'
    items:
    - key: config_yaml_file
      path: config.yaml
    - key: config_yaml_file
      path: config1.yaml
    - key: config_yaml_file
      path: config2.yaml
    - key: config_yaml_file
      path: config3.yaml
    - key: config_yaml_file
      path: config4.yaml
    - key: config_yaml_file
      path: config5.yaml
    - key: config_yaml_file
      path: config6.yaml
    - key: config_yaml_file
      path: config7.yaml
    - key: config_yaml_file
      path: config8.yaml
    - key: config_yaml_file
      path: config9.yaml
    - key: config_yaml_file
      path: config10.yaml
```

ends up with 
```
nobody@dev2-uswest2adevc:/who$ for i in $(ls /who/config/); do echo $i; cat /who/config/$i; done
config10.yaml
ldap:
    username: svc-aas@ad.yelpcorp.com
    password: REDACTED
    ca_cert: /etc/ssl/certs/ca-certificates.crt
    url: ldap://auth.ad.yelpcorp.com:389
    trace: 0
    retry_max: 10
    retry_delay: 2
    debug: 0
    timeout: 3
    base_dn: OU=Yelp,DC=ad,DC=yelpcorp,DC=com
    users_ou: OU=Users,OU=Yelp,DC=ad,DC=yelpcorp,DC=com
    disabled_ou: OU=DisabledUsers,OU=Yelp,DC=ad,DC=yelpcorp,DC=com

workday:
    username: ISU-Engineering
    password: "REDACTED"
config1.yaml
cat: /who/config/config1.yaml: Is a directory
config2.yaml
cat: /who/config/config2.yaml: Is a directory
config3.yaml
cat: /who/config/config3.yaml: Is a directory
config4.yaml
cat: /who/config/config4.yaml: Is a directory
config5.yaml
cat: /who/config/config5.yaml: Is a directory
config6.yaml
cat: /who/config/config6.yaml: Is a directory
config7.yaml
cat: /who/config/config7.yaml: Is a directory
config8.yaml
cat: /who/config/config8.yaml: Is a directory
config9.yaml
cat: /who/config/config9.yaml: Is a directory
config.yaml
cat: /who/config/config.yaml: Is a directory
```

Gonna switch back to just getting a temp directory and using os.makedirs (which I originally had, but thought tempfile.TemporaryDirectory was going to bet better for some reason)